server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -2777,9 +2774,7 @@ private SnapshotsInProgress updatedSnapshotsInProgress(ClusterState currentState
                 for (IndexId indexIdToRefresh : newIndexIdsToRefresh) {
                     updatedIndexIds.put(indexIdToRefresh, new IndexId(indexIdToRefresh.getName(), UUIDs.randomBase64UUID()));
                 }
-                for (int i = 0; i < snapshotEntries.size(); i++) {
-                    snapshotEntries.set(i, snapshotEntries.get(i).withUpdatedIndexIds(updatedIndexIds));
-                }
+                snapshotEntries.replaceAll(e -> e.withUpdatedIndexIds(updatedIndexIds));
```

Comment:
Depending on the size of the entries, running this with a lambda might be quite a bit slower.

Reply:
Possibly, although `ArrayList.replaceAll` has got an implementation that uses indexes rather than a list iterator. Then the only additional cost is calling the lambda - which I expect the JVM would be able to fully inline anyway, as the lambda closure object is not escaping outside this method.

If this is performance sensitive I can revert this, but then should probably put a comment saying this needs to be kept as-is.

Reply:
I think it wouldn't be too hard to write a small JMH that will verify if the lambda is slower. I think your version of the code is much nicer, but I'm suspicious of why the code was written as before.

Reply:
I'll see if I can do a quick check

Reply:
`replaceAll` is actually quicker:
```
Benchmark.testIterator    thrpt    5  375361.849 ±  7966.389  ops/s
Benchmark.testReplaceAll  thrpt    5  443378.984 ± 14116.645  ops/s
```

Reply:
Great! TIL.